C/services/common/include/python_plugin_handle.h

```diff
@@ -17,19 +17,91 @@
 #include <plugin_handle.h>
 #include <Python.h>
 
+typedef void* (*pluginResolveSymbolFn)(const char *);
+typedef void (*pluginCleanupFn)();
+
 /**
- * The PythonPluginHandle class is used to represent an interface to 
+ * The PythonPluginHandle class is the base class used to represent an interface to
  * a plugin that is available as a python script
  */
 class PythonPluginHandle : public PluginHandle
 {
 	public:
-		PythonPluginHandle(const char *name, const char *path);
-		~PythonPluginHandle();
-		void *GetInfo();
-		void *ResolveSymbol(const char* sym);
+		// Base constructor
+		PythonPluginHandle(const char *name, const char *path) : m_name(name) {};
+		/**
+		 * Base destructor
+		 *    - Call cleanup on python plugin interface
+		 *    - Close python plugin interface library handle
+		 */
+		~PythonPluginHandle()
+		{
+			pluginCleanupFn cleanupFn =
```

Comment:
Ensure that "m_hndl" is not NULL, before using it.

Reply:
Adding a check ...